it/controllers/benefits/reimbursed/OtherBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,499 @@
+        }
+
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      "redirect to cya page when the reimbursedCostsVouchersAndNonCashQuestion value is false" in {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        insertCyaData(employmentUserData(isPrior = false, cyaModel(Some(benefits(emptyReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+        urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      result.status shouldBe SEE_OTHER
+      result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+    }
+
+      "redirect to cya page when the benefitsReceived value is false" in {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        insertCyaData(employmentUserData(isPrior = false, cyaModel(benefitsWithNoBenefitsReceived)), userRequest)
+        urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      result.status shouldBe SEE_OTHER
+      result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+    }
+
+  }
+
+  ".submit" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        import Selectors._
+        import user.commonExpectedResults._
+
+        "should render the amount page with empty value error text when there is no input" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+            urlPost(pageUrl(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), body = Map[String, String]())
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption)
+          textOnPageCheck(previousExpectedContent, previousAmountTextSelector)
+          textOnPageCheck(amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("", inputSelector)
+
```

Comment:
can add inputFieldCheck here

Reply:
as discussed not needed as the inputFieldValueCheck would fail if its not there